compiler/AST/symbol.cpp

```diff
@@ -303,6 +308,82 @@ bool VarSymbol::isParameter() const {
   return hasFlag(FLAG_PARAM) || immediate;
 }
 
+
+std::string VarSymbol::docsDirective() {
+  std::string result;
+  if (fDocsTextOnly) {
+    result = "";
+  } else {
+    // TODO: If this is a type (i.e. this->hasFlag(FLAG_TYPE_VARIABLE)), use
+    //       ".. type:: " as directive. (thomasvandoren, 2015-02-21)
+    if (this->isField) {
+      result = ".. attribute:: ";
+    } else {
+      result = ".. data:: ";
+    }
+  }
+  return this->hasFlag(FLAG_CONFIG) ? result + "config " : result;
+}
+
+
+void VarSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC) || this->hasFlag(FLAG_SUPER_CLASS)) {
+      return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  if (this->hasFlag(FLAG_TYPE_VARIABLE)) {
+    *file << "type ";
+  } else if (this->isConstant()) {
+    *file << "const ";
+  } else if (this->isParameter()) {
+    *file << "param ";
+  } else {
+    *file << "var ";
+  }
+
+  *file << this->name;
+
+  if (this->defPoint->exprType != NULL) {
+    *file << ": ";
+    this->defPoint->exprType->prettyPrint(file);
+  }
+
+  // FIXME: Do we want this for non-class globals/configs?
+  //        (thomasvandoren, 2015-02-21)
```

Comment:
The role of the type alias feature is to provide "a type alias" not a brand new type.  For example, if you wrote:

type age = int;

proc foo(x: age); {
  writeln("In age version");
}

proc foo(x: int) {
  writeln("In int version");
}

var myage: age;
var i: int;

foo(myage);
foo(i);

You do not get a cool program in which ints and ages are handled differently, you get an ambiguity because you've written two functions that both take ints.  What that says to me is that I have to know what this type is in order to be able to use it effectively.  If you want a black box type that the user cannot reason about at all, you should write a record and make everything about it private (i.e., you want an ADT not a type alias) or you should propose adding a new structural subtyping feature to the language.  But as the language is defined today, printing out type aliases without what they are aliasing is going to do more good than harm.

I don't buy the type aliases at module and record/class scope should be treated differently argument any more than module and record/class members should be treated differently.